code/datums/mind.dm

```diff
@@ -251,18 +252,26 @@ datum/mind
 		return FALSE
 
 	/// Attempts to remove existing antagonist datums of ID role_id from this mind.
-	proc/remove_antagonist(role_id)
-		for (var/datum/antagonist/A as anything in src.antagonists)
-			if (A.id == role_id)
-				A.remove_self(TRUE)
-				src.antagonists.Remove(A)
-				if (!length(src.antagonists) && src.special_role == A.id)
-					src.special_role = null
-					ticker.mode.traitors.Remove(src)
-					ticker.mode.Agimmicks.Remove(src)
-				qdel(A)
-				src.current.antagonist_overlay_refresh(TRUE, FALSE)
-				return TRUE
+	proc/remove_antagonist(role)
+		var/list/datum/antagonist/antagonist_roles = list()
+		if (istext(role))
+			for (var/datum/antagonist/A as anything in src.antagonists)
+				if (A.id == role)
+					antagonist_roles += A
```

Comment:
```suggestion
					antagonist_roles += A
					break
```

Reply:
See comment about more than one antagonist role existing in parallel.